modules/core/src/main/scala/com.fortysevendeg.hood/model/Benchmark.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.fortysevendeg.hood.model
+
+import io.circe._
+import io.circe.generic.semiauto._
+import cats.syntax.either._
+
+sealed trait BenchmarkMode
+case object Throughput     extends BenchmarkMode
+case object AverageTime    extends BenchmarkMode
+case object SampleTime     extends BenchmarkMode
+case object SingleShotTime extends BenchmarkMode
+case object All            extends BenchmarkMode
```

Comment:
I like this classification but I'd prefer not hardcoding the strings in the decoder and encoder. 

We can use enumeratum with circe integration (https://github.com/lloydmeta/enumeratum#circe):

```scala
import enumeratum._
import enumeratum.EnumEntry._

sealed trait BenchmarkMode extends EnumEntry with Camelcase

object BenchmarkMode extends Enum[BenchmarkMode] with CirceEnum[BenchmarkMode] {

  val values = findValues

  case object Throughput     extends BenchmarkMode
  case object AverageTime    extends BenchmarkMode
  case object SampleTime     extends BenchmarkMode
  case object SingleShotTime extends BenchmarkMode
  case object All            extends BenchmarkMode

}
```

this will give us the decoder and encoder for free. Not sure if we need to use `Camelcase` or `LowerCamelcase` (see https://github.com/lloydmeta/enumeratum#mixins-to-override-the-name)

Reply:
Ops, too late

Reply:
Outdated :disappointed:

Reply:
@fedefernandez @AdrianRaFo thought it was kind of overkill to have all that. Would you agree on that?

Reply:
yes